### PR TITLE
Parse git credential configuration

### PR DESCRIPTION
**Pull request by kassio**
Related to: #490

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -54,7 +54,13 @@ func (s *ConfigSuite) TestUnmarshal(c *C) {
...
 		defaultBranch = main
 [url "ssh://git@github.com/"]
-	insteadOf = https://github.com/
+		insteadOf = https://github.com/
+[credential]
+		helper = osxkeychain
+		helper = "!f() { echo 'password=multiple-helpers'; }; f"
+		helper = "/bin/get-my-git-password"
+[credential "https://kernel.org"]
+		username = foo
```
Comment (MxKaras) [inline config/config_test.go:63]: Say the config file did not have a default entry i.e. only this
```
[credential "https://kernel.org"]
		username = foo
```

What would be populated for `Credentials["default"]`? Should there be a default entry in this case?

File: config/config_test.go [hunk 2]
```diff
@@ -89,6 +95,15 @@ func (s *ConfigSuite) TestUnmarshal(c *C) {
...
 	c.Assert(cfg.Init.DefaultBranch, Equals, "main")
+	c.Assert(cfg.Credentials["default"].Helper, DeepEquals, []string{
+		"osxkeychain",
+		"!f() { echo 'password=multiple-helpers'; }; f",
+		"/bin/get-my-git-password"})
+	c.Assert(cfg.Credentials["default"].Username, Equals, "")
+	c.Assert(cfg.Credentials["default"].UseHttpPath, Equals, false)
+	c.Assert(cfg.Credentials["https://kernel.org"].Helper, DeepEquals, []string{})
+	c.Assert(cfg.Credentials["https://kernel.org"].Username, Equals, "foo")
+	c.Assert(cfg.Credentials["https://kernel.org"].UseHttpPath, Equals, false)
```
Comment (pjbgf) [inline config/config_test.go:106]: Add a case in which `UseHttpPath` is `true`.

**Review by pjbgf — COMMENTED** [review]
@kassio please rebase after addressing the few nits below. Then ping me for a review.

**Comment by kassio** [comment]
Unfortunately, I won't be able to finish this. If anyone wants/can take over it I appreciate.

**Pull request closed without merging**